astronomer/providers/google/cloud/hooks/dataproc.py

```diff
@@ -1,35 +1,115 @@
 import warnings
+from abc import ABC
 from typing import Any, Optional, Sequence, Tuple, Union
 
-from airflow.providers.google.cloud.hooks.dataproc import DataprocHook
+from airflow.providers.google.common.consts import CLIENT_INFO
 from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
-from asgiref.sync import sync_to_async
+from google.api_core import gapic_v1
+from google.api_core.client_options import ClientOptions
 from google.api_core.retry import Retry
-from google.cloud.dataproc_v1 import Job
+from google.cloud.dataproc_v1 import (
+    ClusterControllerAsyncClient,
+    Job,
+    JobControllerAsyncClient,
+)
+from google.cloud.dataproc_v1.types import clusters
 
 JobType = Union[Job, Any]
 
 
-class DataprocHookAsync(DataprocHook):
-    """
-    Async Hook for Google Cloud Dataproc APIs.
-    All the methods in the hook where project_id is used must be called with
-    keyword arguments rather than positional.
-    """
+class DataprocHookAsync(GoogleBaseHook, ABC):
+    """Async Hook for Google Cloud Dataproc APIs"""
+
+    def get_cluster_client(
+        self, region: Optional[str] = None, location: Optional[str] = None
+    ) -> ClusterControllerAsyncClient:
+        """
+        Get async cluster controller client for GCP Dataproc.
+
+        :param region: The Cloud Dataproc region in which to handle the request.
+        :param location: (To be deprecated). The Cloud Dataproc region in which to handle the request.
+        """
+        if location is not None:
```

Comment:
what about moving this block to a method and then calling that method here because this block is repeating at multiple places?

Reply:
Done